### ISSUE TITLE
LuaGlueDebug.h doesn't work on window/visual studio

### ISSUE THREAD

**Issue by guanlisheng** [Tomasu/LuaGlue#16]

**Comment by Tomasu** [comment]
Thanks for the report. I'm going to need more information than that. I don't use MSVC.

**Comment by guanlisheng** [comment]
one is LG_debug

**Comment by Tomasu** [comment]
I still need the actual error messages. but I may have fixed it. let me know.

**Comment by JMLX42** [comment]
Here is a fix:

``` cpp
#ifndef LUAGLUE_DEBUG_H_GUARD
#define LUAGLUE_DEBUG_H_GUARD

#ifdef LUAGLUE_DEBUG

#include <cstring>
static inline void LG_Debug_(const char *FILE, const char *FUNCTION, int LINE, const char *t, const char *format, ...)
{
    va_list ap;
    va_start(ap, format);
    int msg_len = vsnprintf(0, 0, format, ap);
    va_end(ap);

    char *msg = new char[msg_len+1];
    if(!msg)
        return;

    msg[msg_len] = 0;

    va_start(ap, format);
    vsnprintf(msg, (size_t)msg_len+1, format, ap);
    va_end(ap);

    const char *fptr = strrchr(FILE, '/');
    if(!fptr)
        fptr = strrchr(FILE, '\\');

    if(!fptr)
        fptr = FILE;
    else
        fptr++;

    printf("%s:%s:%i: %s: %s\n", fptr, FUNCTION, LINE, t, msg);

    delete[] msg;
}

#define LG_Debug(a, b...) LG_Debug_(__FILE__, __PRETTY_FUNCTION__, __LINE__, "D", a, ##b)
#define LG_Warn(a, b...) LG_Debug_(__FILE__, __PRETTY_FUNCTION__, __LINE__, "W", a, ##b)
#define LG_Error(a, b...) LG_Debug_(__FILE__, __PRETTY_FUNCTION__, __LINE__, "E", a, ##b)

#else

#define LG_Debug(a, ...) ((void)a)
#define LG_Warn(a, ...) ((void)a)
#define LG_Error(a, ...) ((void)a)

#endif /* !defined DEBUG */

#endif /* LUAGLUE_DEBUG_H_GUARD */
```

**Comment by Tomasu** [comment]
Hmm, I'm not sure this is correct. I seem to recall that the msvcrt.dll vsnprintf doesn't support returning the length if you pas null as the buffer. a fixed sized buffer would probably have to be used.

**Comment by Tomasu** [comment]
I've applied a fix in 365c901b81a0583b2a4367ac6219b939dda80f44 that might help here.
